### PR TITLE
Fix spammy GetHostIP error log in kubelet

### LINKED ISSUE #8571: Kubelet repeatedly logs "Cannot get host IP: cannot get node: node <name> not found"
From https://groups.google.com/d/msg/google-containers/WPGJsSmGV6g/Ux5Q22veyEAJ

Hi there,

hope you can help me. Our kubelets post the following error multiple times per minute :

E0520 10:34:48.111760    1360 kubelet.go:1973] Cannot get host IP: host IP unknown; known addresses: []

E0520 10:34:58.108223    1360 kubelet.go:1973] Cannot get host IP: host IP unknown; known addresses: []

They are running within the AWS cloud and the hostnames of the machine are also specified within the /etc/hosts file. In effect i have no idea why the kublet is not able to get its host ip. By the way the api-server and kublets are behind firewalls(security groups), do i need to open some special ports? Nevertheless the communication between the different parts of kubernetes seems to work, very smooth. So what do i miss? Many thanks in advance.

Regards
Stefan

### PR DESCRIPTION
**Pull request by mikedanese**
Fixes #8571

### CODE REVIEW

**Comment by dchen1107** [comment]
If we really hit GetHostIP error due to misconfigured minions in a OSS cluster, shouldn't this change hide the issue and make the debugging much harder? cc/ to @roberthbailey

**Comment by mikedanese** [comment]
@dchen1107 please take another look.

**Comment by dchen1107** [comment]
@jdef This PR breaks mesos on initializing kubelet.

File: contrib/mesos/pkg/executor/service/service.go [hunk 1]
```diff
@@ -327,6 +327,7 @@ func (ks *KubeletExecutorServer) createAndInitKubelet(
 		gcPolicy,
 		pc.SeenAllSources,
 		kc.RegisterNode,
+		kc.StandaloneMode,
```
Comment (jdef) [inline contrib/mesos/pkg/executor/service/service.go]: :+1:

**Comment by jdef** [comment]
Can you add a placeholder in the KubeletConfig initialization? Even if it's just a comment (like some of the others), it's helpful to keep track of all these config options.
- https://github.com/mikedanese/kubernetes/blob/host-ip-spam/contrib/mesos/pkg/executor/service/service.go#L211

**Comment by mikedanese** [comment]
@jdef how do I run the build with make? I ran make test test_integration and it passed but shippable still broke.

**Comment by davidopp** [comment]
@jdef I thought we had not enabled building the contrib/mesos stuff yet. Specifically, in #8882, the changes you made to build/common.sh and hack/lib/golang.sh, I thought one has to set 
KUBERNETES_CONTRIB env var to cause this stuff to build. I thought we were waiting to enable that until the remaining upstreaming PRs were submitted. What is causing it to build right now in Shippable?

**Comment by jdef** [comment]
I think @dchen1107 was calling out the k8s-mesos breaking changes if one were to enable building the mesos extensions. By default they are not enabled.

Setting `KUBERNETES_CONTRIB=mesos` is required to build k8s-mesos components. however, until #9265 is merged the mesos component build won't work.

File: contrib/mesos/pkg/executor/service/service.go [hunk 2]
```diff
@@ -218,14 +218,15 @@ func (s *KubeletExecutorServer) Run(hks hyperkube.Interface, _ []string) error {
...
-		RegisterNode:                   s.RegisterNode,
+		PodInfraContainerImage:  s.PodInfraContainerImage,
+		SyncFrequency:           s.SyncFrequency,
+		RegistryPullQPS:         s.RegistryPullQPS,
+		RegistryBurst:           s.RegistryBurst,
+		MinimumGCAge:            s.MinimumGCAge,
+		MaxPerPodContainerCount: s.MaxPerPodContainerCount,
+		MaxContainerCount:       s.MaxContainerCount,
+		RegisterNode:            s.RegisterNode,
+		// StandaloneMode: false
```
Comment (jdef) [inline contrib/mesos/pkg/executor/service/service.go]: :+1:

**Comment by dchen1107** [comment]
LGTM and shippable is back too. Thanks everyone.

**Comment by guenter** [comment]
:+1:

**Comment by dchen1107** [comment]
Close #8571 :-)

**Comment by karlkfi** [comment]
To echo @jdef:

Master currently does not build with `KUBERNETES_CONTRIB=mesos make`. This is expected, because `contrib/mesos/cmd/k8sm-redirfd` depends on `contrib/mesos/cmd/k8sm-controller-manager`, which is a part of #9265, which was split out to make review easier.

However, `make` should work fine (ignores contrib/mesos).

**Comment by roberthbailey** [comment]
LGTM.

**Comment by mikedanese** [comment]
Jenkins won't stop building!

**Comment by jdef** [comment]
It just loves your code.

**Comment by ixdy** [comment]
I really wish the GitHub PR plugin for Jenkins was smarter about recognizing that it doesn't need to build/test an obsolete commit.

**Comment by mikedanese** [comment]
@ixdy does the PR builder accept PRs?

**Comment by ixdy** [comment]
@mikedanese it's on GitHub, so likely: https://github.com/jenkinsci/ghprb-plugin

**Pull request merged**